controllers/idler/idler_controller_test.go

```diff
@@ -438,14 +506,256 @@ func TestEnsureIdlingFailed(t *testing.T) {
 				// then
 				require.EqualError(t, err, fmt.Sprintf("failed to ensure idling 'alex-stage': %s", errMsg))
 				assert.Equal(t, reconcile.Result{}, res)
-				memberoperatortest.AssertThatIdler(t, idler.Name, cl).HasConditions(memberoperatortest.FailedToIdle(errMsg))
+				memberoperatortest.AssertThatIdler(t, idler.Name, cl).ContainsCondition(memberoperatortest.FailedToIdle(errMsg))
 			}
 
 			assertCanNotDeleteObject(&appsv1.DaemonSet{}, "can't delete daemonset")
 			assertCanNotDeleteObject(&batchv1.Job{}, "can't delete job")
 			assertCanNotDeleteObject(&corev1.Pod{}, "can't delete pod")
 		})
 	})
+
+	t.Run("Fail if cannot update notification creation failed status", func(t *testing.T) {
+		// given
+		idler := &toolchainv1alpha1.Idler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "john-dev",
+				Labels: map[string]string{
+					toolchainv1alpha1.OwnerLabelKey: "john",
+				},
+			},
+			Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 30},
+		}
+		namespaces := []string{"dev", "stage"}
+		usernames := []string{"john"}
+		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "john", "advanced", "abcde11", namespaces, usernames)
+		reconciler, req, cl, _ := prepareReconcile(t, idler.Name, getHostCluster, idler, nsTmplSet) // not adding mur
+		idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
+		preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo)
+
+		// when
+		// first reconcile to start tracking pods
+		_, err := reconciler.Reconcile(context.TODO(), req)
+
+		// second reconcile to delete pods and create notification
+		cl.MockStatusUpdate = func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+			return fmt.Errorf("cannot set status to fail")
+		}
+		_, err = reconciler.Reconcile(context.TODO(), req)
+
+		// then
+		// since no mur, error should have been wrapped as status update, but status update fails
+		require.EqualError(t, err, "failed to ensure idling 'john-dev': cannot set status to fail")
```

Comment:
This just made me realize that we are returning an error if the status update fails and that would also stop the pod tracking so we should just log the error in this case too.

Reply:
done in [f347b00](https://github.com/codeready-toolchain/member-operator/pull/357/commits/f347b0044fdde5e0f21331cdc292c5ba5ff8b409)